src/sagemaker/serve/builder/transformers_builder.py

```diff
@@ -109,7 +119,7 @@ def _get_hf_metadata_create_model(self) -> Type[Model]:
         """
 
         hf_model_md = get_huggingface_model_metadata(
-            self.model, self.env_vars.get("HUGGING_FACE_HUB_TOKEN")
+            self.env_vars.get("HF_MODEL_ID"), self.env_vars.get("HUGGING_FACE_HUB_TOKEN")
```

Comment:
what happens if `model` is set with an HF model ID but the environment variable doesn't exist?

Reply:
Yes, HF Model ID gets set before the line is executed. This is where it is called:
[https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/serve/builder/transformers_builder.py#L248-L261](url)
[https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/serve/builder/transformers_builder.py#L263](url)
[https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/serve/builder/transformers_builder.py#L83](url)